### PR TITLE
AutoFollow Stats API Addition

### PR DESCRIPTION
**Pull request by gbbafna**
Signed-off-by: Gaurav Bafna <gbbafna@amazon.com>

### Description
An API to list out all auto follow jobs and its success, failure counters
 
### Issues Resolved
https://github.com/opensearch-project/cross-cluster-replication/issues/130

### Sample Request , Response
```
➜  scripts curl -XGET  "${FOLLOWER}/_plugins/_replication/autofollow_stats" -d'{}'  -H 'Content-Type: application/json'
{
  "num_success_start_replication" : 3,
  "num_failed_start_replication" : 0,
  "num_failed_leader_calls" : 0,
  "failed_indices" : [ ],
  "autofollow_stats" : [
    {
      "name" : "my-replication",
      "pattern" : "leader-*",
      "num_success_start_replication" : 2,
      "num_failed_start_replication" : 0,
      "num_failed_leader_calls" : 0,
      "failed_indices" : [ ]
    },
    {
      "name" : "my-replication-2",
      "pattern" : "lead-*",
      "num_success_start_replication" : 1,
      "num_failed_start_replication" : 0,
      "num_failed_leader_calls" : 0,
      "failed_indices" : [ ]
    }
  ]
}
```
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by tbhanu-amzn — APPROVED** [review]

File: src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt [hunk 1]
```diff
@@ -55,9 +60,10 @@ class AutoFollowTask(id: Long, type: String, action: String, description: String
     private var trackingIndicesOnTheCluster = setOf<String>()
     private var failedIndices = ConcurrentSkipListSet<String>() // Failed indices for replication from this autofollow task
```
Comment (saikaranam-amazon) [inline src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt:61]: Can we add failed Indices to response?

File: src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt [hunk 2]
```diff
@@ -55,9 +60,10 @@ class AutoFollowTask(id: Long, type: String, action: String, description: String
...
     private var retryScheduler: Scheduler.ScheduledCancellable? = null
-    private var failureCount = 0
+    lateinit var stat: AutoFollowStat
```
[the remark this answers is not in the document]
Comment (gbbafna) [reply src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt:63]: We can't as replicationMetadata is only initialized later on before `execute` is called.

File: src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt (hunk 1 above)
Comment (gbbafna) [reply src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt:61]: Good idea. Added it .

**Review by tbhanu-amzn — APPROVED** [review]

**Review by tbhanu-amzn — APPROVED** [review]

**Review by saikaranam-amazon — APPROVED** [review]
Can we have the issue open for test failures to investigate?

**Comment by gbbafna** [comment]
> Can we have the issue open for test failures to investigate?

yes, opened https://github.com/opensearch-project/cross-cluster-replication/issues/155 for same.

**Pull request merged**